src/components/utils/src/timer.cc

```diff
@@ -64,6 +64,8 @@ timer::Timer::~Timer() {
   StopDelegate();
   single_shot_ = true;
 
+  thread_->set_delegate(NULL);
```

Comment:
@vlantonov you don't need to call `set_delegate(NULL)` explicitly. It will be done by `ThreadDelegate` destructor